besu/src/test/java/org/hyperledger/besu/services/BesuEventsImplTest.java

```diff
@@ -144,24 +144,29 @@ public void syncStatusEventDoesNotFireAfterUnsubscribe() {
 
   @Test
   public void newBlockEventFiresAfterSubscribe() {
-    final AtomicReference<BlockHeader> result = new AtomicReference<>();
+    final AtomicReference<BlockPropagated> result = new AtomicReference<>();
     serviceImpl.addBlockPropagatedListener(result::set);
-
+    final Block block = generateBlock();
     assertThat(result.get()).isNull();
-    blockBroadcaster.propagate(generateBlock(), UInt256.of(1));
+    blockBroadcaster.propagate(block, UInt256.of(1));
 
     assertThat(result.get()).isNotNull();
+    assertThat(result.get().getBlockHeader()).isEqualTo(block.getHeader());
+    assertThat(result.get().getTotalDifficulty()).isEqualTo(UInt256.of(1));
   }
 
   @Test
   public void newBlockEventDoesNotFireAfterUnsubscribe() {
-    final AtomicReference<BlockHeader> result = new AtomicReference<>();
+    final AtomicReference<BlockPropagated> result = new AtomicReference<>();
     final long id = serviceImpl.addBlockPropagatedListener(result::set);
 
     assertThat(result.get()).isNull();
-    blockBroadcaster.propagate(generateBlock(), UInt256.of(1));
+    final Block block = generateBlock();
+    blockBroadcaster.propagate(block, UInt256.of(1));
```

Comment:
By changing the `UInt256` value, you can also be asserting the value is getting assigned correctly.
```suggestion
    blockBroadcaster.propagate(block, UInt256.of(2));
```